group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/GroupCoordinatorRuntimeMetrics.java

```diff
@@ -208,8 +209,8 @@ public void recordEventQueueTime(long durationMs) { }
     public void recordEventQueueProcessingTime(long durationMs) { }
 
     @Override
-    public void recordThreadIdleRatio(double ratio) {
-        threadIdleRatioSensor.record(ratio);
+    public synchronized void recordThreadIdleTime(long idleTimeMs, long currentTimeMs) {
+        threadIdleTimeRate.record(metrics.config(), idleTimeMs, currentTimeMs);
```

Comment:
I am confused here. Shouldn't we call `record` on the sensor?

Reply:
A sensor requires both "total" and "rate" metrics. Since the "total idle time" metric is more or less useless, I have removed it. But this requires us to directly instantiate the Rate object.

The alternative would be to revert it to a Sensor and also create a "total" metric.

Reply:
Hum… Can’t we really use a Rate in a Sensor? This seems strange to me.

Reply:
Thanks for the correction. Added the rate to the sensor.